### PR TITLE
Composite In L3 Test procedure documentation

### PR DESCRIPTION
**Pull request by akhilbhas**
Test procedure documentation for dsCompositeIn L3 testcases.

### CODE REVIEW

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+# dscompositeIn HAL L3 Python Test Procedure
+
+## Table of Contents
+
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [Setting Up Test Environment](#setting-up-test-environment)
+- [Streams Required](#streams-required)
+- [Test Setup Connections](#test-setup-connections)
+- [Test Cases](#test-cases)
```
Comment (Anbukannadhasan) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:9]: Can you add all the test case names in the table of contents

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+  - Question: "Connect the compositeIn source device to Port port_type and press Y: (Y/N)"
+  - Press **Y** if user connected the device and acknowledge the same. (this will mark the step as PASS).
+  - Press **N** if user could not connect the device. (this will mark the step as FAIL).
+
+- Device connect Confirmation:
+
+  - Test will check if the event has reached the device. 
+  - If event detected will mark the step as PASS
+  - If event not detected will mark the step as DAIL
```
Comment (Anbukannadhasan) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:219]: DAIL instead of FAIL. Typo. Please check through out the file

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+- Sink
+
+#### User Input Required - test02
+
+**Yes**: User interaction is necessary to connect/disconnect Composite Source device (This will be automated later).
+
+#### Acceptance Criteria - test02
+
+Test detects the status of the Source ddevice connected to Composite In port
```
Comment (Anbukannadhasan) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:254]: typo ddevice

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+- Device Connect prompt:
+
+    The test will Request User to connect Source device to Composite In port:
+
+  - Question: "Connect the compositeIn source device to Port port_type and press Y: (Y/N)"
+  - Press **Y** if user connected the device and acknowledge the same. (this will mark the step as PASS).
+  - Press **N** if user could not connect the device. (this will mark the step as FAIL).
+
+- Device changes the scale and request user to confirm:
```
Comment (Anbukannadhasan) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:382]: This test is not related to scale

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+
+## Setting Up Test Environment
+
+To execute `HAL` `L3` Python test cases, need a Python environment. Follow these steps mentioned in [HPK Public Documentation](https://github.com/rdkcentral/rdk-hpk-documentation/blob/main/README.md)
+
+### Update Configuration Files
+
+#### Rack Configuration File
+
+Example Rack configuration File: `ut/host/tests/configs/example_rack_config.yml`
```
Comment (srinivasgtl) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:31]: Do we need to have a right link here?

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+      defaults: &defaults
+        artifacts:
+          - "<path>/bin/hal_test"
+          - "<path>/bin/libut_control.so"
+          - "<path>/bin/Sink_compositeIn.yaml"
+          - "<path>/bin/run.sh"
+        execute:
+          - "chmod +x /opt/HAL/dscompositeIn_L3/hal_test"
+          - "chmod +x /opt/HAL/dscompositeIn_L3/run.sh"
+          - cp -rf /usr/lib/libdshal.so /opt/HAL/dscompositeIn_L3/
```
Comment (srinivasgtl) [inline docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md:133]: should the cp command go with quotes here? ""

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 1 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 2 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 3 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 4 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 5 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

File: docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md (hunk 6 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md]: done

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by srinivasgtl — APPROVED** [review]
ship it.

**Comment by thanushreevani** [comment]
this has been included in pull request https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/210
